lib/src/migrators/module.dart

```diff
@@ -46,54 +35,55 @@ class ModuleMigrator extends MigratorBase {
   /// Global functions defined at any time during the migrator run.
   final _functions = normalizedMap<FunctionRule>();
 
-  /// Local variables, mixins, and functions for migrations in progress.
-  ///
-  /// The migrator will modify this as it traverses stylesheets. When at the
-  /// top level of a stylesheet, this will be null.
-  LocalScope _localScope;
-
-  /// Current stylesheet being actively migrated.
-  StylesheetMigration get _currentMigration =>
-      _activeMigrations.isNotEmpty ? _activeMigrations.last : null;
-
-  /// The patches to be aplied to the stylesheet being actively migrated.
-  List<Patch> get patches => _currentMigration.patches;
-
   /// Runs the module migrator on [entrypoint] and its dependencies and returns
   /// a map of migrated contents.
   ///
-  /// If [_migrateDependencies] is false, the migrator will still be run on
+  /// If [migrateDependencies] is false, the migrator will still be run on
   /// dependencies, but they will be excluded from the resulting map.
-  @override
-  p.PathMap<String> migrateFile(String entrypoint) {
-    var migration = _migrateStylesheet(entrypoint, Directory.current.path);
-    var results = p.PathMap<String>();
-    addMigration(StylesheetMigration migration) {
-      var migrated = migration.migratedContents;
-      if (migrated != migration.contents) {
-        results[migration.path] = migrated;
-      }
-    }
-
-    if (_migrateDependencies) {
-      _migrations.values.forEach(addMigration);
-    } else {
-      addMigration(migration);
+  void migrateFile(String entrypoint) {
+    _ModuleMigrationVisitor(this, entrypoint).run();
+    if (!migrateDependencies) {
+      migrated.removeWhere((path, contents) => path != entrypoint);
     }
-    return results;
   }
+}
 
-  /// Migrates the stylesheet at [path] if it hasn't already been migrated and
-  /// returns the StylesheetMigration instance for it regardless.
-  StylesheetMigration _migrateStylesheet(String path, String directory) {
-    path = canonicalizePath(p.join(directory, path));
-    return _migrations.putIfAbsent(path, () {
-      var migration = StylesheetMigration(path);
-      _activeMigrations.add(migration);
-      visitStylesheet(migration.stylesheet);
-      _activeMigrations.remove(migration);
-      return migration;
-    });
+class _ModuleMigrationVisitor extends MigrationVisitor {
+  final ModuleMigrator migrator;
+  final String path;
+
+  _ModuleMigrationVisitor(this.migrator, this.path);
+
+  _ModuleMigrationVisitor newInstance(String newPath) =>
+      _ModuleMigrationVisitor(migrator, newPath);
+
+  /// Namespaces of modules used in this stylesheet.
+  final namespaces = p.PathMap<String>();
```

Comment:
I'd keep these members private. Even though the class itself is private now, making its members private as well helps communicate that they aren't intended to be used outside this class (technically they *can* be, but readers use a private member as a mental shortcut that indicates class privacy).